src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1692 @@
+
+  If this function is called in ARK_FULLRHS_START or ARK_FULLRHS_END mode and
+  evaluating the RHS functions is necessary, we store the vector f(t,y) in Fe[0]
+  for reuse in the first stage of the subsequent time step.
+
+  In ARK_FULLRHS_END mode we check if the method is "stiffly accurate" and, if
+  appropriate, copy the vector F[stages - 1] to F[0] for reuse in the first
+  stage of the subsequent time step.
+
+  ARK_FULLRHS_OTHER mode is only called for dense output in-between steps, or
+  when estimating the initial time step size, so we strive to store the
+  intermediate parts so that they do not interfere with the other two modes.
+  ----------------------------------------------------------------------------*/
+int lsrkStep_FullRHS(ARKodeMem ark_mem, sunrealtype t, N_Vector y, N_Vector f,
+                     int mode)
+{
+  int retval;
+  ARKodeLSRKStepMem step_mem;
+
+  /* access ARKodeLSRKStepMem structure */
+  retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* perform RHS functions contingent on 'mode' argument */
+  switch (mode)
+  {
+  case ARK_FULLRHS_START:
+
+    /* compute the RHS */
+    if (!(ark_mem->fn_is_current))
+    {
+      retval = step_mem->fe(t, y, step_mem->Fe[0], ark_mem->user_data);
+      step_mem->nfe++;
+      if (retval != 0)
+      {
+        arkProcessError(ark_mem, ARK_RHSFUNC_FAIL, __LINE__, __func__, __FILE__,
+                        MSG_ARK_RHSFUNC_FAILED, t);
+        return (ARK_RHSFUNC_FAIL);
+      }
+    }
+
+    /* copy RHS vector into output */
+    N_VScale(ONE, step_mem->Fe[0], f);
+
+    break;
+
+  case ARK_FULLRHS_END:
+    /* No further action is needed since the currently
+    available methods evaluate the RHS at the end of each time step*/
+
+    break;
+
+  case ARK_FULLRHS_OTHER:
+
+    /* call f */
+    retval = step_mem->fe(t, y, f, ark_mem->user_data);
+    step_mem->nfe++;
+    if (retval != 0)
+    {
+      arkProcessError(ark_mem, ARK_RHSFUNC_FAIL, __LINE__, __func__, __FILE__,
+                      MSG_ARK_RHSFUNC_FAILED, t);
+      return (ARK_RHSFUNC_FAIL);
+    }
+
+    break;
+
+  default:
+    /* return with RHS failure if unknown mode is passed */
+    arkProcessError(ark_mem, ARK_RHSFUNC_FAIL, __LINE__, __func__, __FILE__,
+                    "Unknown full RHS mode");
+    return (ARK_RHSFUNC_FAIL);
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_TakeStepRKC:
+
+  This routine serves the primary purpose of the LSRKStepRKC module:
+  it performs a single RKC step (with embedding, if possible).
+
+  The output variable dsmPtr should contain estimate of the
+  weighted local error if an embedding is present; otherwise it
+  should be 0.
+
+  The input/output variable nflagPtr is used to gauge convergence
+  of any algebraic solvers within the step.  As this routine
+  involves no algebraic solve, it is set to 0 (success).
```

Comment:
You "resolved" this comment prematurely.  Since `lsrkStep_TakeStepRKC` might return with `*nflagPtr = ARK_RETRY_STEP` then the current comment right here is incorrect.  It looks like you set **both** `*nflagPtr` and the return value to `ARK_RETRY_STEP` when an insufficient number of stages is allowed for the current step size.  Do you really need to do both, or would just one of these two be sufficient?  

In any case, the comment in this function header should accurately represent what occurs in the function, so if you do need to set `*nflagPtr` to `ARK_RETRY_STEP`, then this sentence ("The input/output variable nflagPtr ...") needs to be fixed.